src/parsers/duration.rs

```diff
@@ -1,81 +1,70 @@
-
-use nom::digit;
 use super::super::Duration;
-
+use nom::bytes::complete::tag;
+use nom::character::complete::digit1;
+use nom::combinator::{map_res, opt};
+use nom::sequence::{terminated, Tuple};
+use nom::IResult;
 use std::str;
-use std::str::FromStr;
-
-named!(u32_digit<u32>,
```

Comment:
This construct would generate a function that would take a byte array and operate on it. Since we're passed a string from the user of the library, just switch to using `&str` throughout, which eliminates some of the mapping we previously did like this: `str::from_utf8`